src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -151,7 +151,9 @@ public static function register($mode = false)
                 }
 
                 if ('weak' !== $mode && ($deprecations['unsilenced'] || $deprecations['remaining'] || $deprecations['other'])) {
-                    exit(1);
+                    register_shutdown_function(function () {
```

Comment:
what would be even better would be to defer the whole close, so that deprecations triggered at shutdown time are also caught (and add a related test of course :) )

Reply:
The downside being that if a shutdown function uses `exit` and the whole closure is deferred too, then you won't get any deprecations. But I think catching shutdown time deprecations might outweight this, what do you think? If we want to go crazy, we may even reset the state after the first, non deferred display and if any deprecations are caught while shutting down, then display them in the deferred closure.

Reply:
as you, both are fine to me :)

Reply:
I'll see what I can manage

Reply:
Ended up "going crazy", tell me what you think